Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm

```diff
@@ -434,7 +434,7 @@ static String commandNameForSelectorName(const String& selectorName)
     
     bool didPerformAction = false;
     
-    if (!frame->settings().eventHandlerDrivenSmoothKeyboardScrollingEnabled()) {
+    if (!frame->settings().eventHandlerDrivenSmoothKeyboardScrollingEnabled() || frame->eventHandler().focusedScrollableAreaUsesScrollSnap()) {
```

Comment:
I think this can adopt your new combo getter, right?

Reply:
Oops, missed that. Will fix!